File: summary/func_struct.go
package main
import "fmt"

type Product struct {
  Name, Note string
  Price      int
}

func main() {
  p := Product{Name: "milk", Note: "yammy", Price: 600}
  name, price := add(p)
  fmt.Println(name, price)
  fmt.Println(name)
  fmt.Println(price)
}

func add(p Product) (int, string) {
  return p.Price+11, p.Name
}

// go run func_struct.go
// 611 milk
// 611
// milk

File: summary/func_struct.swift
struct Product {
	var Name, Note :String
    var Price      :Int
}

func add(product: Product) -> (sum:Int, msg:String) {
    return (product.Price+11, product.Name)
}

let product = Product(Name: "milk", Note: "yammy", Price: 600)
var test = add(product: product)

print(test)
print(test.sum)
print(test.msg)

// swift func_struct.swift
// (sum: 711, msg: "milk")
// 611
// milk